components/EmbraceModal.vue

```diff
@@ -0,0 +1,176 @@
+  <ModalWithBackdrop v-model="showModal">
+    <div class="pt-28 mb-10 max-w-2xl h-128 max-h-full text-slate-800 text-left">
+      <form class="justify-center items-center bg-sushi-100 flex flex-col px-10 rounded-2xl max-md:px-5">
+        <div
+          class="justify-between items-stretch self-stretch flex gap-5 mt-10 max-md:max-w-full max-md:flex-wrap max-md:mt-10"
+        >
+          <div class="text-2xl font-bold grow shrink basis-auto max-md:max-w-full">
+            {{ embracePage?.data.preview_title || 'Email Preview' }}
+          </div>
+        </div>
+        <div class="self-stretch text-left text-xl tracking-wide mt-10 max-md:max-w-full max-md:mt-6">
+          <PrismicRichText
+            :field="embracePage?.data.preview_description"
+            class="text-md mb-4 whitespace-pre-wrap text-gray-800"
+            fallback="If you agree with the information below, click on 'Go To Your Mail App'. You may also edit the message body beforehand."
+          />
+        </div>
+        <div class="items-center self-stretch flex w-full justify-between gap-5 mt-6 max-md:max-w-full max-md:flex-wrap">
+          <div class="text-left text-xl font-medium leading-6 grow shrink basis-auto my-auto max-md:max-w-full">
+            {{ embracePage?.data.preview_area_label || 'Generated message' }}
+          </div>
+        </div>
+        <div
+          id="previewBox"
+          class="items-stretch self-stretch border border-[color:var(--brand-green-medium,#7BB123)] bg-white flex flex-col mt-4 p-2 rounded-2xl border-solid max-md:max-w-full max-md:px-5"
+        >
+          <div
+            name="addressFields"
+            class="justify-center items-stretch bg-slate-800 bg-opacity-10 flex flex-col px-3 py-2 rounded-lg max-md:max-w-full"
+          >
+            <EmailPreviewField
+              name="toField"
+              title="To: "
+              :value="form.bankEmail"
+            />
+            <EmailPreviewField
+              name="bccField"
+              title="BCC: "
+              :value=" embracePage?.data.bcc_email || 'embrace@bank.green'"
+            />
+            <EmailPreviewField
+              v-if="form?.email"
+              name="fromField"
+              title="From: "
+              :value="form.email"
+            />
+            <EmailPreviewField
+              name="subjectField"
+              title="Subject: "
+              :value="form?.subject || 'Error retrieving subject line'"
+            />
+          </div>
+          <TextField
+            v-model="messageBody"
+            class="col-span-full w-full text-gray-800"
+            rows="10"
+            name="embraceText"
+            type="text"
+            :value="form?.body"
+            dark
+          />
+        </div>
+        <div
+          class="button-white justify-center items-center mt-4 bg-white self-stretch flex gap-1 py-2 rounded-lg border border-gray-800 border-solid max-md:px-5"
+          @click.prevent.stop="copyText"
```

Comment:
after a user successfully copied the text, I think it would be good we maybe add a checkmark at the end of the button or the copy icon changes to a checkmark icon so that a user gets a reaction from their action

Reply:
Could additionally change the button text ('Copied!').  
 
If we indicate this, I think there should also be a listener/event so when they modify the message it would reset the button.